discovery_engine_flutter/lib/src/assets/asset_copier.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import 'dart:io' show Directory, File;
+import 'dart:typed_data' show ByteData;
+
+import 'package:flutter/services.dart' show rootBundle;
+import 'package:xayn_discovery_engine/discovery_engine.dart'
+    show Manifest, kAssetsPath, logger;
+
+/// A signature for a function loading the assets from bundle.
+typedef AssetLoader = Future<ByteData> Function(String key);
+
+class FlutterBundleAssetCopier {
+  final String bundleAssetsPath;
+  final String appDir;
+  final AssetLoader _loadAsset;
+
+  FlutterBundleAssetCopier({
+    required this.appDir,
+    required this.bundleAssetsPath,
+    AssetLoader? loadAsset,
+  })  : assert(appDir.isNotEmpty),
+        assert(bundleAssetsPath.isNotEmpty),
+        _loadAsset = loadAsset ?? rootBundle.load;
```

Comment:
>  it's weird that dart doesn't allow for direct assignment of private fields in an optional named argument list oO so i'd say up to you if you want to change it.

It's not that weird, if it's private it should not affect the public interface.

Reply:
not true, for example in the following it's perfectly fine to have a private field in a public interface, the limitation only exists for optional private arguments and hence is quite inconsistent.
```dart
class MyClass {
  final int _myField;
  MyClass(this._myField);
}
```

Reply:
Because that will not break the public interface. In this case you can rename `_myField` and all the code will continue to work, this does not holds for named parameters.